File: backend/routes/projects.py
"""
Projects API Routes - Platform-wide project management.

Projects are shared workspaces that can be accessed by multiple agents and team members.
Uses SQLAlchemy models for persistence.
"""

from flask import Blueprint, request, jsonify, g
from datetime import datetime
import uuid

from core.auth import require_auth, user_can_access_project, user_can_manage_project_settings
from core.database import db
from core.models import Project, Team, TeamMember
from core.project_settings import ProjectSettings, PROJECT_SETTING_KEYS

projects_bp = Blueprint('projects', __name__)


def get_user_team(user_email: str) -> Team:
    """Get user's team, creating one if needed."""
    member = TeamMember.query.filter_by(user_id=user_email).first()
    if member:
        return member.team

    # Create new team with user as owner
    team_id = str(uuid.uuid4())
    team = Team(
        team_id=team_id,
        owner_id=user_email,
        name=f"{user_email.split('@')[0]}'s Team"
    )
    db.session.add(team)

    member = TeamMember(
        member_id=str(uuid.uuid4()),
        team_id=team_id,
        user_id=user_email,
        name=user_email.split('@')[0],
        role='owner'
    )
    db.session.add(member)
    db.session.commit()
    return team


def get_user_projects(user_email: str):
    """Get all projects for a user (via their team)."""
    team = get_user_team(user_email)
    return Project.query.filter_by(team_id=team.team_id).all()


@projects_bp.route('/api/projects', methods=['GET'])
@require_auth
def list_projects():
    """List all projects for the current user."""
    user_email = g.user_id
    projects = get_user_projects(user_email)
    return jsonify({'success': True, 'projects': [p.to_dict() for p in projects]})


@projects_bp.route('/api/projects', methods=['POST'])
@require_auth
def create_project():
    """Create a new project."""
    user_email = g.user_id
    data = request.get_json()
    name = data.get('name', '').strip()
    description = data.get('description', '').strip()
    # Default to all agents if none specified
    agents = data.get('agents') or [
        'marketResearch', 'salesHelper', 'contentMarketing',
        'communityNetwork', 'eventNetworking', 'executiveAssistant', 'dataInsights'
    ]

    if not name:
        return jsonify({'error': 'Project name is required'}), 400

    team = get_user_team(user_email)
    project = Project(
        project_id=str(uuid.uuid4()),
        team_id=team.team_id,
        owner_id=user_email,
        name=name,
        description=description,
    )
    project.agents = agents
    project.data = {}

    db.session.add(project)
    db.session.commit()
    return jsonify({'success': True, 'project': project.to_dict()})


@projects_bp.route('/api/projects/<project_id>', methods=['GET'])
@require_auth
def get_project(project_id):
    """Get a specific project by ID."""
    if not user_can_access_project(g.user_id, project_id):
        return jsonify({'error': 'Project not found'}), 404

    project = Project.query.filter_by(project_id=project_id).first()
    return jsonify({'success': True, 'project': project.to_dict()})


@projects_bp.route('/api/projects/<project_id>', methods=['PUT'])
@require_auth
def update_project(project_id):
    """Update a project."""
    if not user_can_access_project(g.user_id, project_id):
        return jsonify({'error': 'Project not found'}), 404

    project = Project.query.filter_by(project_id=project_id).first()

    data = request.get_json()
    if 'name' in data:
        project.name = data['name'].strip()
    if 'description' in data:
        project.description = data['description'].strip()
    if 'agents' in data:
        project.agents = data['agents']
    if 'status' in data and data['status'] in ['active', 'archived', 'completed']:
        project.status = data['status']
    if 'data' in data:
        current = project.data
        current.update(data['data'])
        project.data = current

    project.updated_at = datetime.utcnow()
    db.session.commit()
    return jsonify({'success': True, 'project': project.to_dict()})


@projects_bp.route('/api/projects/<project_id>', methods=['DELETE'])
@require_auth
def delete_project(project_id):
    """Delete a project (owner only)."""
    project = Project.query.filter_by(project_id=project_id).first()
    if not project or not user_can_access_project(g.user_id, project_id):
        return jsonify({'error': 'Project not found'}), 404
    if project.owner_id != g.user_id:
        return jsonify({'error': 'Only project owner can delete'}), 403

    db.session.delete(project)
    db.session.commit()
    return jsonify({'success': True, 'message': 'Project deleted'})


@projects_bp.route('/api/projects/<project_id>/data', methods=['PUT'])
@require_auth
def update_project_data(project_id):
    """Update agent-specific data within a project."""
    if not user_can_access_project(g.user_id, project_id):
        return jsonify({'error': 'Project not found'}), 404

    data = request.get_json()
    agent_key = data.get('agent')
    agent_data = data.get('data', {})

    if not agent_key:
        return jsonify({'error': 'Agent key is required'}), 400

    project = Project.query.filter_by(project_id=project_id).first()
    if agent_key not in project.agents:
        return jsonify({'error': f'Agent {agent_key} not enabled for this project'}), 403

    project.set_agent_data(agent_key, agent_data)
    project.updated_at = datetime.utcnow()
    db.session.commit()
    return jsonify({'success': True, 'project': project.to_dict()})


@projects_bp.route('/api/projects/<project_id>/settings', methods=['GET'])
@require_auth
def get_project_settings(project_id):
    """
    Get this project's AI provider settings. Visible to anyone with
    project access (so members know which key their work is billed to),
    but see PUT below for who can change it.
    """
    if not user_can_access_project(g.user_id, project_id):
        return jsonify({'error': 'Project not found'}), 404

    settings = ProjectSettings().list(project_id, include_values=True)
    return jsonify({
        'success': True,
        'settings': settings,
        'canManage': user_can_manage_project_settings(g.user_id, project_id),
    })


@projects_bp.route('/api/projects/<project_id>/settings', methods=['PUT'])
@require_auth
def update_project_settings(project_id):
    """Set a project-level AI provider key. Owner/admin only - this key
    takes priority over every member's personal key for AI actions taken
    inside this project (see core/ai_client.py)."""
    if not user_can_access_project(g.user_id, project_id):
        return jsonify({'error': 'Project not found'}), 404
    if not user_can_manage_project_settings(g.user_id, project_id):
        return jsonify({'error': 'Only the project owner or a team admin can change project settings'}), 403

    data = request.get_json() or {}
    key = data.get('key')
    value = data.get('value')
    if key not in PROJECT_SETTING_KEYS:
        return jsonify({'error': f'Unsupported setting key: {key}'}), 400
    if not value:
        return jsonify({'error': 'value is required'}), 400

    ProjectSettings().set(project_id, key, value)
    return jsonify({'success': True, 'message': f'Project setting {key} saved'})


@projects_bp.route('/api/projects/<project_id>/settings/<key>', methods=['DELETE'])
@require_auth
def delete_project_setting(project_id, key):
    """Remove a project-level AI provider key. Owner/admin only - after
    this, AI actions in the project fall back to each member's personal
    key (or the platform default)."""
    if not user_can_access_project(g.user_id, project_id):
        return jsonify({'error': 'Project not found'}), 404
    if not user_can_manage_project_settings(g.user_id, project_id):
        return jsonify({'error': 'Only the project owner or a team admin can change project settings'}), 403

    deleted = ProjectSettings().delete(project_id, key)
    if not deleted:
        return jsonify({'error': 'Setting not found'}), 404
    return jsonify({'success': True, 'message': f'Project setting {key} deleted'})

File: frontend/src/projects/Projects.js
import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import './Projects.css';
import { showToast } from '../core/toast';
import Header from '../core/Header';
import { authJsonHeaders, authOptionalHeaders } from '../core/authHeaders';
import { BackButton, showConfirm } from '../components';
import { AGENTS } from '../config/agentsConfig';
import { initializeDemoProjects } from '../hooks/useProjectData';
import { useMode } from '../contexts';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:5000';
const PROJECTS_STORAGE_KEY = 'enableAgentsProjects';

// Helper to get/save projects from localStorage
const getStoredProjects = () => {
  try {
    const data = localStorage.getItem(PROJECTS_STORAGE_KEY);
    return data ? JSON.parse(data) : [];
  } catch {
    return [];
  }
};

const saveStoredProjects = (projects) => {
  try {
    localStorage.setItem(PROJECTS_STORAGE_KEY, JSON.stringify(projects));
  } catch (e) {
    console.warn('Failed to save projects:', e);
  }
};

// Initialize demo projects on first load
initializeDemoProjects();

const Icons = {
  ArrowLeft: () => (
    <svg viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
      <path d="M19 12H5M12 19l-7-7 7-7"/>
    </svg>
  ),
  Plus: () => (
    <svg viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
      <line x1="12" y1="5" x2="12" y2="19"/><line x1="5" y1="12" x2="19" y2="12"/>
    </svg>
  ),
  Folder: () => (
    <svg viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
      <path d="M22 19a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h5l2 3h9a2 2 0 0 1 2 2z"/>
    </svg>
  ),
  Trash: () => (
    <svg viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
      <polyline points="3 6 5 6 21 6"/><path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"/>
    </svg>
  ),
  X: () => (
    <svg viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
      <line x1="18" y1="6" x2="6" y2="18"/><line x1="6" y1="6" x2="18" y2="18"/>
    </svg>
  ),
  Users: () => (
    <svg viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
      <path d="M17 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"/><circle cx="9" cy="7" r="4"/><path d="M23 21v-2a4 4 0 0 0-3-3.87"/><path d="M16 3.13a4 4 0 0 1 0 7.75"/>
    </svg>
  ),
  Settings: () => (
    <svg viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
      <circle cx="12" cy="12" r="3"/>
      <path d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 1 1-2.83 2.83l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-4 0v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 1 1-2.83-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1 0-4h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 1 1 2.83-2.83l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 4 0v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 1 1 2.83 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 0 4h-.09a1.65 1.65 0 0 0-1.51 1z"/>
    </svg>
  ),
};

const AI_KEY_FIELDS = [
  { key: 'openai_key', label: 'OpenAI API Key', placeholder: 'sk-...', type: 'password' },
  { key: 'anthropic_key', label: 'Anthropic API Key', placeholder: 'sk-ant-...', type: 'password' },
  {
    key: 'preferred_model', label: 'Preferred Model', type: 'select',
    options: [
      { value: 'gpt-4o-mini', label: 'GPT-4o Mini (Fast, Cheap)' },
      { value: 'gpt-4o', label: 'GPT-4o (Powerful)' },
      { value: 'claude-3-haiku', label: 'Claude 3 Haiku (Fast)' },
      { value: 'claude-3-sonnet', label: 'Claude 3 Sonnet (Balanced)' },
    ],
  },
];

function Projects() {
  const navigate = useNavigate();
  const userEmail = localStorage.getItem('userEmail') || '';
  const { isDemoMode } = useMode();

  const [projects, setProjects] = useState([]);
  const [loading, setLoading] = useState(true);
  const [showCreateModal, setShowCreateModal] = useState(false);
  const [newProject, setNewProject] = useState({
    name: '',
    description: '',
    team_id: '',
  });
  const [creating, setCreating] = useState(false);
  const [teams, setTeams] = useState([]);
  const [loadingTeams, setLoadingTeams] = useState(false);

  // Project AI key settings
  const [settingsProject, setSettingsProject] = useState(null);
  const [projectSettings, setProjectSettings] = useState(null);
  const [canManageSettings, setCanManageSettings] = useState(false);
  const [settingsLoading, setSettingsLoading] = useState(false);
  const [settingsEditValues, setSettingsEditValues] = useState({});
  const [settingsSavingKey, setSettingsSavingKey] = useState('');

  useEffect(() => {
    fetchProjects();
  }, [isDemoMode]);

  // Fetch teams when create modal opens
  useEffect(() => {
    if (showCreateModal && !isDemoMode) {
      fetchTeams();
    }
  }, [showCreateModal, isDemoMode]);

  const fetchTeams = async () => {
    setLoadingTeams(true);
    try {
      const res = await fetch(`${API_URL}/api/team`, {
        headers: authOptionalHeaders(),
      });
      if (res.ok) {
        const data = await res.json();
        setTeams(data.members ? [{ id: 'default', name: 'My Team' }] : []);
        // If there's a default team, select it
        if (data.team_id) {
          setNewProject(prev => ({ ...prev, team_id: data.team_id }));
        }
      }
    } catch (err) {
      console.error('Failed to fetch teams:', err);
    } finally {
      setLoadingTeams(false);
    }
  };

  const fetchProjects = async () => {
    if (isDemoMode) {
      // Use localStorage for demo mode
      const storedProjects = getStoredProjects();
      setProjects(storedProjects);
      setLoading(false);
      return;
    }

    try {
      const res = await fetch(`${API_URL}/api/projects`, {
        headers: authOptionalHeaders(),
      });
      if (res.ok) {
        const data = await res.json();
        setProjects(data.projects || []);
      }
    } catch (err) {
      console.error('Failed to fetch projects:', err);
    } finally {
      setLoading(false);
    }
  };

  const handleCreate = async () => {
    if (!newProject.name.trim()) {
      showToast('Please enter a project name', 'warning');
      return;
    }

    // All agents are enabled by default
    const allAgents = ['marketResearch', 'salesHelper', 'contentMarketing',
      'communityNetwork', 'eventNetworking', 'executiveAssistant', 'dataInsights'];

    if (isDemoMode) {
      const project = {
        id: `proj-${Date.now()}`,
        name: newProject.name,
        description: newProject.description,
        agents: allAgents,
        owner: userEmail || 'demo@example.com',
        status: 'active',
        createdAt: new Date().toISOString().split('T')[0],
        updatedAt: new Date().toISOString(),
        data: {},
      };
      const updatedProjects = [...projects, project];
      setProjects(updatedProjects);
      saveStoredProjects(updatedProjects);
      setNewProject({ name: '', description: '', team_id: '' });
      setShowCreateModal(false);
      showToast('Project created', 'success');
      return;
    }

    setCreating(true);
    try {
      const res = await fetch(`${API_URL}/api/projects`, {
        method: 'POST',
        headers: authJsonHeaders(),
        body: JSON.stringify(newProject),
      });

      if (res.ok) {
        showToast('Project created', 'success');
        setNewProject({ name: '', description: '', team_id: '' });
        setShowCreateModal(false);
        fetchProjects();
      } else {
        const data = await res.json();
        showToast(data.error || 'Failed to create', 'error');
      }
    } catch {
      showToast('Failed to create project', 'error');
    } finally {
      setCreating(false);
    }
  };

  const handleDelete = async (projectId) => {
    const project = projects.find(p => p.id === projectId);
    const confirmed = await showConfirm({
      title: 'Delete project?',
      message: `This will permanently remove "${project?.name || 'this project'}" and its agent data. This cannot be undone.`,
      confirmLabel: 'Delete',
      cancelLabel: 'Cancel',
      variant: 'danger',
    });
    if (!confirmed) return;

    if (isDemoMode) {
      const updatedProjects = projects.filter(p => p.id !== projectId);
      setProjects(updatedProjects);
      saveStoredProjects(updatedProjects);
      showToast('Project deleted', 'success');
      return;
    }

    try {
      const res = await fetch(`${API_URL}/api/projects/${projectId}`, {
        method: 'DELETE',
        headers: authOptionalHeaders(),
      });

      if (res.ok) {
        showToast('Project deleted', 'success');
        fetchProjects();
      }
    } catch {
      showToast('Failed to delete project', 'error');
    }
  };

  const getAgentName = (agentId) => {
    return AGENTS[agentId]?.name || agentId;
  };

  const openProjectSettings = async (project) => {
    setSettingsProject(project);
    setSettingsEditValues({});
    setSettingsLoading(true);
    try {
      const res = await fetch(`${API_URL}/api/projects/${project.id}/settings`, {
        headers: authOptionalHeaders(),
      });
      const data = await res.json();
      if (res.ok && data.success) {
        setProjectSettings(data.settings?.ai?.settings || {});
        setCanManageSettings(Boolean(data.canManage));
      } else {
        showToast(data.error || 'Failed to load project settings', 'error');
        setProjectSettings({});
        setCanManageSettings(false);
      }
    } catch {
      showToast('Failed to load project settings', 'error');
      setProjectSettings({});
      setCanManageSettings(false);
    } finally {
      setSettingsLoading(false);
    }
  };

  const closeProjectSettings = () => {
    setSettingsProject(null);
    setProjectSettings(null);
    setSettingsEditValues({});
  };

  const handleSaveProjectSetting = async (key) => {
    const value = settingsEditValues[key];
    if (!value || !value.trim()) return;

    setSettingsSavingKey(key);
    try {
      const res = await fetch(`${API_URL}/api/projects/${settingsProject.id}/settings`, {
        method: 'PUT',
        headers: authJsonHeaders(),
        body: JSON.stringify({ key, value: value.trim() }),
      });
      const data = await res.json();
      if (res.ok && data.success) {
        showToast('Project setting saved', 'success');
        setSettingsEditValues(prev => {
          const next = { ...prev };
          delete next[key];
          return next;
        });
        openProjectSettings(settingsProject);
      } else {
        showToast(data.error || 'Failed to save setting', 'error');
      }
    } catch {
      showToast('Failed to save setting', 'error');
    } finally {
      setSettingsSavingKey('');
    }
  };

  const handleRemoveProjectSetting = async (key) => {
    setSettingsSavingKey(key);
    try {
      const res = await fetch(`${API_URL}/api/projects/${settingsProject.id}/settings/${key}`, {
        method: 'DELETE',
        headers: authOptionalHeaders(),
      });
      const data = await res.json();
      if (res.ok && data.success) {
        showToast('Project setting removed', 'success');
        openProjectSettings(settingsProject);
      } else {
        showToast(data.error || 'Failed to remove setting', 'error');
      }
    } catch {
      showToast('Failed to remove setting', 'error');
    } finally {
      setSettingsSavingKey('');
    }
  };

  const Icon = ({ name }) => {
    const I = Icons[name];
    return I ? <I /> : null;
  };

  return (
    <div className="projects-page">
      <Header />
      <div className="projects-container">
        <header className="projects-header">
          <BackButton />
          <div className="header-content">
            <h1>Projects</h1>
            <p className="text-muted">
              Shared workspaces across agents and team members
            </p>
          </div>
          <button className="btn-create" onClick={() => setShowCreateModal(true)}>
            <Icon name="Plus" />
            New Project
          </button>
        </header>

        {loading ? (
          <div className="loading">Loading...</div>
        ) : projects.length === 0 ? (
          <div className="empty-state-card">
            <Icon name="Folder" />
            <h3>No projects yet</h3>
            <p>Create a project to share work across agents and team members.</p>
            <button className="btn-primary" onClick={() => setShowCreateModal(true)}>
              Create First Project
            </button>
          </div>
        ) : (
          <div className="projects-grid">
            {projects.map(project => (
              <div key={project.id} className="project-card">
                <div className="project-card-header">
                  <div className="project-icon">
                    <Icon name="Folder" />
                  </div>
                  <div className="project-title">
                    <h3>{project.name}</h3>
                    <span className={`status-badge ${project.status}`}>
                      {project.status}
                    </span>
                  </div>
                  {!isDemoMode && (
                    <button
                      className="btn-icon"
                      onClick={() => openProjectSettings(project)}
                      title="AI provider settings"
                    >
                      <Icon name="Settings" />
                    </button>
                  )}
                  <button
                    className="btn-icon danger"
                    onClick={() => handleDelete(project.id)}
                    title="Delete project"
                  >
                    <Icon name="Trash" />
                  </button>
                </div>

                {project.description && (
                  <p className="project-description">{project.description}</p>
                )}

                <div className="project-meta">
                  <span className="meta-item">
                    Created {project.createdAt}
                  </span>
                  {project.data && Object.keys(project.data).length > 0 && (
                    <div className="project-stats">
                      {project.data.tasks && <span>{project.data.tasks} tasks</span>}
                      {project.data.leads && <span>{project.data.leads} leads</span>}
                      {project.data.content && <span>{project.data.content} content</span>}
                      {project.data.attendees && <span>{project.data.attendees} attendees</span>}
                    </div>
                  )}
                </div>

                <div className="project-actions">
                  <button
                    className="btn-open"
                    onClick={() => {
                      // Navigate to first enabled agent with this project
                      const firstAgent = project.agents[0];
                      const route = AGENTS[firstAgent]?.route;
                      if (route) {
                        navigate(`${route}?project=${project.id}`);
                      }
                    }}
                  >
                    Open Project
                  </button>
                </div>
              </div>
            ))}
          </div>
        )}
      </div>

      {/* Create Project Modal */}
      {showCreateModal && (
        <div className="modal-overlay" onClick={() => setShowCreateModal(false)}>
          <div className="modal modal-lg" onClick={e => e.stopPropagation()}>
            <div className="modal-header">
              <h3>Create New Project</h3>
              <button className="btn-icon" onClick={() => setShowCreateModal(false)}>
                <Icon name="X" />
              </button>
            </div>
            <div className="modal-body">
              <div className="field">
                <label>Project Name *</label>
                <input
                  type="text"
                  value={newProject.name}
                  onChange={(e) => setNewProject({ ...newProject, name: e.target.value })}
                  placeholder="e.g., Q3 Marketing Campaign"
                  autoFocus
                />
              </div>
              <div className="field">
                <label>Description</label>
                <textarea
                  value={newProject.description}
                  onChange={(e) => setNewProject({ ...newProject, description: e.target.value })}
                  placeholder="Brief description of the project"
                  rows={2}
                />
              </div>
              {!isDemoMode && (
                <div className="field">
                  <label>Team</label>
                  <div className="team-select-row">
                    <select
                      value={newProject.team_id}
                      onChange={(e) => setNewProject({ ...newProject, team_id: e.target.value })}
                      disabled={loadingTeams}
                    >
                      <option value="">Select team...</option>
                      {teams.map(team => (
                        <option key={team.id} value={team.id}>{team.name}</option>
                      ))}
                    </select>
                    <button
                      type="button"
                      className="btn-link"
                      onClick={() => navigate('/team')}
                    >
                      Manage Team
                    </button>
                  </div>
                </div>
              )}
              <p className="field-hint" style={{ marginTop: '8px', color: 'var(--color-text-muted)' }}>
                All agents will have access to this project.
              </p>
            </div>
            <div className="modal-footer">
              <button className="btn-secondary" onClick={() => setShowCreateModal(false)}>
                Cancel
              </button>
              <button
                className="btn-primary"
                onClick={handleCreate}
                disabled={!newProject.name.trim() || creating}
              >
                {creating ? 'Creating...' : 'Create Project'}
              </button>
            </div>
          </div>
        </div>
      )}

      {/* Project AI Settings Modal */}
      {settingsProject && (
        <div className="modal-overlay" onClick={closeProjectSettings}>
          <div className="modal modal-lg" onClick={e => e.stopPropagation()}>
            <div className="modal-header">
              <h3>AI Settings — {settingsProject.name}</h3>
              <button className="btn-icon" onClick={closeProjectSettings}>
                <Icon name="X" />
              </button>
            </div>
            <div className="modal-body">
              {settingsLoading ? (
                <div className="loading">Loading...</div>
              ) : (
                <>
                  <p className="field-hint" style={{ marginBottom: '12px', color: 'var(--color-text-muted)' }}>
                    {canManageSettings
                      ? 'Set a key here and every AI action inside this project uses it instead of a member\'s personal key or the platform default.'
                      : 'Only the project owner or a team admin can change these keys. Shown below is what this project currently uses.'}
                  </p>
                  {AI_KEY_FIELDS.map(field => {
                    const setting = projectSettings?.[field.key] || {};
                    const isEditing = field.key in settingsEditValues;
                    const isSaving = settingsSavingKey === field.key;
                    return (
                      <div className="field" key={field.key}>
                        <label>
                          {field.label}
                          {setting.configured && (
                            <span className="status-badge active" style={{ marginLeft: '8px' }}>
                              {field.type === 'select' ? `Set to ${setting.value}` : 'Configured'}
                            </span>
                          )}
                        </label>
                        {field.type === 'select' ? (
                          <div className="project-setting-row">
                            <select
                              value={isEditing ? settingsEditValues[field.key] : (setting.value || setting.default || field.options[0].value)}
                              onChange={(e) => setSettingsEditValues(prev => ({ ...prev, [field.key]: e.target.value }))}
                              disabled={!canManageSettings}
                            >
                              {field.options.map(opt => (
                                <option key={opt.value} value={opt.value}>{opt.label}</option>
                              ))}
                            </select>
                            {canManageSettings && isEditing && (
                              <button
                                type="button"
                                className="btn-primary"
                                onClick={() => handleSaveProjectSetting(field.key)}
                                disabled={isSaving}
                              >
                                {isSaving ? 'Saving...' : 'Save'}
                              </button>
                            )}
                          </div>
                        ) : (
                          canManageSettings && (
                            <div className="project-setting-row">
                              <input
                                type="password"
                                placeholder={setting.configured ? setting.value : field.placeholder}
                                value={settingsEditValues[field.key] || ''}
                                onChange={(e) => setSettingsEditValues(prev => ({ ...prev, [field.key]: e.target.value }))}
                              />
                              <button
                                type="button"
                                className="btn-primary"
                                onClick={() => handleSaveProjectSetting(field.key)}
                                disabled={isSaving || !settingsEditValues[field.key]?.trim()}
                              >
                                {isSaving ? 'Saving...' : 'Save'}
                              </button>
                              {setting.configured && (
                                <button
                                  type="button"
                                  className="btn-secondary"
                                  onClick={() => handleRemoveProjectSetting(field.key)}
                                  disabled={isSaving}
                                >
                                  Remove
                                </button>
                              )}
                            </div>
                          )
                        )}
                        {!canManageSettings && !setting.configured && (
                          <p className="field-hint">Not configured — falls back to each member's personal key or the platform default.</p>
                        )}
                      </div>
                    );
                  })}
                </>
              )}
            </div>
            <div className="modal-footer">
              <button className="btn-secondary" onClick={closeProjectSettings}>Close</button>
            </div>
          </div>
        </div>
      )}
    </div>
  );
}

export default Projects;
